ledger/src/mask/mask_impl.rs

```diff
@@ -1087,6 +1112,32 @@ impl BaseLedger for MaskImpl {
         set
     }
 
+    fn token_owner(&self, token_id: TokenId) -> Option<AccountId> {
+        let (parent, token_owners) = match self {
+            Root { database, .. } => return database.token_owner(token_id),
+            Attached {
+                parent,
+                token_owners,
+                ..
+            } => (Some(parent), token_owners),
+            Unattached { token_owners, .. } => (None, token_owners),
+        };
+
+        if let Some(account_id) = token_owners.get(&token_id).cloned() {
+            return Some(account_id);
+        };
+
+        parent.as_ref()?.token_owner(token_id)
+    }
+
+    fn token_owners(&self) -> HashSet<AccountId> {
+        // TODO: Not sure if it's the correct impl
```

Comment:
this returns every account so probably not correct (not sure what it is supposed to do, maybe just accounts that own custom tokens?)